src/repository.c

```diff
@@ -2085,7 +2086,18 @@ int git_repository_init_ext(
 		/* TODO: reinitialize the templates */
 	} else {
 		if ((error = repo_init_structure(repo_path.ptr, wd, opts)) < 0 ||
-		    (error = repo_init_config(repo_path.ptr, wd, opts->flags, opts->mode)) < 0 ||
+		    (error = repo_init_config(repo_path.ptr, wd, opts->flags, opts->mode)) < 0)
+			goto out;
+
+		if ((error = git_buf_joinpath(&head_path, repo_path.ptr, GIT_HEAD_FILE)) < 0)
```

Comment:
Style: I'd merge that `if` with the one above.

(I'm also still worried that every time a `GIT_HEAD_FILE` occurrence is added, it usually means custom refdb won't know about it).

Reply:
> Style: I'd merge that if with the one above.

Fixed

> (I'm also still worried that every time a GIT_HEAD_FILE occurrence is added, it usually means custom refdb won't know about it).

Yup, I thought about that, too. In this case it's fine though, as we're creating a completely new repository with `git_repository_init_ext`, which doesn't support custom backends anyway. So we need to have `GIT_HEAD_FILE` created here. _But_, I think in the long term we should fix `git_repository_create_head`. This one is a generic version and should thus use the refdb.